app/forms/mappings/Formatters.scala

```diff
@@ -106,7 +107,7 @@ trait Formatters {
                                                args: Seq[String] = Seq.empty): Formatter[BigDecimal] =
     new Formatter[BigDecimal] {
       val numericRegexp = """^-?(\-?)(\d*)(\.?)(\d*)$"""
-      val decimalRegexp = """^-?(\d*\.\d{1,2})$"""
+      val decimalRegexp = """^-?(\d*\.\d{2,2})$"""
```

Comment:
`val decimalRegexp = """^-?(\d*\.\d{2,2})$"""` only needs to match exactly 2 times eg 
`val decimalRegexp = """^-?(\d*\.\d{2})$"""`